### ISSUE TITLE
Add Prometheus metrics export to training monitor

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#24]
## Summary

The `monitor` module with trueno-db integration should support Prometheus metrics export for integration with realizar's serve metrics and standard observability stacks.

## Proposed API

```rust
// In src/monitor/prometheus.rs
pub struct PrometheusExporter {
    registry: prometheus::Registry,
}

impl PrometheusExporter {
    pub fn new() -> Self;
    
    /// Export metrics in Prometheus text format
    pub fn export(&self) -> String;
    
    /// Record training metrics
    pub fn record_epoch(&self, epoch: usize, loss: f64, lr: f64);
    pub fn record_batch(&self, batch: usize, loss: f64, throughput: f64);
    pub fn record_validation(&self, epoch: usize, metrics: &ValidationMetrics);
}

// Metrics to export:
// - entrenar_epoch_loss{model="name"}
// - entrenar_learning_rate{model="name"}
// - entrenar_batch_throughput{model="name"}
// - entrenar_validation_accuracy{model="name"}
// - entrenar_training_duration_seconds{model="name"}
```

## Integration Points

- realizar serve metrics (unified dashboard)
- Grafana dashboards
- AlertManager for training anomalies

## Acceptance Criteria

- [ ] PrometheusExporter implemented
- [ ] Feature-gated behind `prometheus` feature
- [ ] Example showing metrics endpoint
- [ ] Documentation in book

**Comment by noahgift** [comment]
Implemented in src/monitor/prometheus/ with PrometheusExporter (exporter.rs), metric types (types.rs), and tests. Exports training metrics in Prometheus text format.
